freshmaker/pyxis.py

```diff
@@ -0,0 +1,252 @@
+        page = 0
+        while True:
+            local_params["page"] = page
+            response_data = self._make_request(entity, params=local_params)
+            # When the page after the actual last page is released, data will be an empty list
+            if not response_data.get('data'):
+                break
+            ret.extend(response_data['data'])
+            page += 1
+
+        return ret
+
+    def get_operator_indices(self, params=None):
+        """
+        Get all index images for organization(s)(configurable) from Pyxis
+
+        :param dict params: parameters to add to GET request
+        """
+        if params is None:
+            request_params = {}
+        else:
+            request_params = copy.deepcopy(params)
+        organization = conf.pyxis_index_image_organization
+        if organization:
+            request_params["filter"] = "organization==" + organization
+
+        return self._pagination("operators/indices", request_params)
+
+    def _get_bundles_per_index_image(self, index_images, params=None):
+        """
+        Get bundle images for all index images
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        if params is None:
+            request_params = {}
+        else:
+            request_params = copy.deepcopy(params)
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest',
+                          'data.bundle_path']
+        request_params['include'] = ','.join(include_fields)
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            request_params['filter'] = f"source_index_container_path=={path}"
+            bundles_per_index_image[path] = self._pagination('operators/bundles',
+                                                             request_params)
+
+        return bundles_per_index_image
+
+    def get_latest_bundles(self, index_images, params={}):
+        """
+        Get latest bundle images per channel per index image
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: latest bundle images per channel per index image
+        :rtype: list
+        """
+        bundles_per_index_image = \
+            self._get_bundles_per_index_image(index_images, params)
+
+        ret_bundles = []
+        for index_image, bundles in bundles_per_index_image.items():
+            bundle_per_channel = {}
+            # get latest versions of bundle images per channel
+            for bundle in bundles:
+                channel = bundle.get('channel_name')
```

Comment:
Will `channel_name` ever not be set? If not, then access the key directly to ensure a `KeyError` is raised in the event Pyxis misbehaves.